pkg/flexvolume/flexvolume.go

```diff
@@ -0,0 +1,308 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flexvolume
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/golang/glog"
+	utiljson "k8s.io/frakti/pkg/util/json"
+	"k8s.io/frakti/pkg/util/knownflags"
+)
+
+type FlexVolumeDriver struct {
+	uuid string
+
+	volId        string
+	fsType       string
+	cinderConfig string
+	readOnly     bool
+	manager      *FlexManager
+
+	linuxMounter *LinuxMounter
+
+	// metadata provides meta of the volume
+	metadata map[string]interface{}
+}
+
+func NewFlexVolumeDriver(uuid string) *FlexVolumeDriver {
+	return &FlexVolumeDriver{
+		uuid:         uuid,
+		linuxMounter: NewLinuxMounter(),
+	}
+}
+
+// Invocation: <driver executable> init
+func (d *FlexVolumeDriver) init() (map[string]interface{}, error) {
+	// "{\"status\": \"Success\", \"capabilities\": {\"attach\": false}}"
```

Comment:
status will be added automatically by "doRun()" for any success cmd (err == nil), see `formatResult()`